File: sast-go/cases/completeness/single_app_tracing/control_flow/conditional_stmt/conditional_switch_004_F/conditional_switch_004_F.go
package conditional_switch_004_F


// evaluation information start
// real case = false
// evaluation item = 完整度->单应用跟踪完整度->流程控制语句->条件语句
// scene introduction = switch init
// level = 2
// bind_url = completeness/single_app_tracing/control_flow/conditional_stmt/conditional_switch_004_F/conditional_switch_004_F
// evaluation information end

func conditional_switch_004_F(__taint_src string) {
	condition := "_"
	switch data := __taint_src; data {
	case condition:
		__taint_sink(condition)
	}
}

func __taint_sink(o interface{}) {}

File: sast-js/case/accuracy/path_sensitive/loop_conditional_stmt/no_solver/conditional_switch_004_F.js
// evaluation information start
// real case = false
// evaluation item = 准确度->路径敏感分析->条件语句、条件表达式和循环结构->无需通过对不同的条件进行求解，即能够区分不同的执行路径的状态
// scene introduction = 区分switch准入条件（不需求解）->switch->区分分支
// level = 3
// bind_url = accuracy/path_sensitive/loop_conditional_stmt/no_solver/conditional_switch_004_F
// evaluation information end

function conditional_switch_004_F(__taint_src) {
  let data = "";
  switch (2) {
    case 1:
      __taint_sink(data);
      break;
    default:
      data = __taint_src;
  }
}

function __taint_sink(o) { }
